### ISSUE TITLE
UMAP - labeling points

### ISSUE THREAD

**Issue by bordumb** [lmcinnes/umap#169]
Hello,

I am using UMAP together with HDBSCAN.

I'm a bit stuck on figuring out how to take the embeddings I've created (the coordinates) and mapping that to my clusters; essentially mapping the output of UMAP to HDBSCAN.

## UMAP to create embeddings - Good
Using code like below to create embeddings
```
neighbor_count = 9
embedding = umap.UMAP(n_neighbors=neighbor_count).fit_transform(data)
embedding
```
And then the embedding looks like this:
```
array([[ 0.1897283, -9.7579365],
       [ 3.2526913, -5.828025 ],
       [-2.722304 ,  5.7153099],
       ...,
       [ 4.501594 ,  6.74332  ],
       [ 5.9605994,  3.2897992],
       [14.772566 , -6.1294155]], dtype=float32)
```

## HDBSCAN to create clusters - Good
```
clusterer = hdbscan.HDBSCAN(min_cluster_size=neighbor_count, gen_min_span_tree=True).fit(embedding)
data['cluster'] = clusterer.labels_
print data[['item_id','cluster']].head()
```
And so now I can map each itemId to a cluster so that it looks something like this:

```
    item_id  cluster
0  111      241
1  222      241
2  333      329
3  444      329
4  555      400
```

## Map item_id/cluster back to embedding - ???
This is where I'm a bit stuck on how UMAP and HDBSCAN work together.

What I'd like is to have a dataset that looks something like:
```
    item_id  cluster  embeddingX  embeddingY
0  111      241      0.1897283      -9.7579365
1  222      241      0.1999283      -9.5579365
2  333      329      5.9605994      3.2897992
3  444      329      5.6059944      3.1897992
4  555      400      4.6059944      3.1897992
```
Any pointers on how to do this?
Or any sample repos with similar work?

I'm essentially trying to do this to help in labeling a visualization I'm building, where I'll treat the embedding as X/Y coordinates and then label each cluster.

**Comment by lmcinnes** [comment]
Everything should be aligned, index-wise (in terms of order, not the actual pandas index) for both umap and hdbscan, so in principle you can just do:

```python
data['embeddingX'] = embedding[:, 0]
data['embeddingY'] = embedding[:, 1]
```

and you will have the dataframe you want.
